lib/usd/pxrUsdPreviewSurface/usdPreviewSurfacePlugin.cpp

```diff
@@ -26,78 +26,73 @@
 #include <maya/MDrawRegistry.h>
 
 #include <pxr/base/tf/envSetting.h>
+#include <pxr/base/tf/stringUtils.h>
 
-PXR_NAMESPACE_USING_DIRECTIVE
+PXR_NAMESPACE_OPEN_SCOPE
 
 namespace {
-const MString _RegistrantId("mayaUsd");
-int _registrationCount = 0;
-
-// Name of the plugin registering the preview surface class.
-MString _registrantPluginName;
-
+TfToken::Set _registeredTypeNames;
 }
-
-PXR_NAMESPACE_OPEN_SCOPE
-
 /* static */
-MStatus
-PxrMayaUsdPreviewSurfacePlugin::initialize(MFnPlugin& plugin)
+MStatus PxrMayaUsdPreviewSurfacePlugin::initialize(
+    MFnPlugin&     plugin,
+    const MString& typeName,
+    MTypeId        typeId,
+    const MString& registrantId)
 {
-    // If we're already registered, do nothing.
-    if (_registrationCount++ > 0) {
-        return MS::kSuccess;
+    TfToken tfTypeName(typeName.asChar());
```

Comment:
Sorry, kind of pedantic, but I might suggest calling this `typeNameToken` or something, just to avoid any potential confusion with this being a `TfType` which is a very different thing.